staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -528,9 +528,9 @@ func completeOpenAPI(config *openapicommon.Config, version *version.Info) {
 	}
 }
 
-// StopNotify returns a lifecycle signal of genericapiserver shutting down.
-func (c *Config) StopNotify() <-chan struct{} {
-	return c.lifecycleSignals.ShutdownInitiated.Signaled()
+// DrainedNotify returns a lifecycle signal of genericapiserver already drained while shutting down.
+func (c *Config) DrainedNotify() <-chan struct{} {
+	return c.lifecycleSignals.InFlightRequestsDrained.Signaled()
```

Comment:
InFlightRequestsDrained is definitely clearer than ShutdownInitiated... is the swap correct for all existing callers / consumers?

Reply:
There is exactly one user of that (this was added within last two weks) being etcd healthcheck, and it's correct there.
So yeah - we should swap - I will split that our into a separate PR, I guess.

Reply:
Split this out to https://github.com/kubernetes/kubernetes/pull/110129